workers/unity/Packages/io.improbable.worker.sdk/ParameterConversion.cs

```diff
@@ -0,0 +1,233 @@
+                    newEvent.Data.AddAll(fields);
+
+                    newItem.Event = newEvent;
+                    break;
+                default:
+                    throw new NotSupportedException("Invalid Item Type provided.");
+            }
+
+            return newItem;
+        }
+
+        public static unsafe void ConvertItem(Item item, ItemCallback callback)
+        {
+            var newItem = Item.GetThreadLocalItem();
+
+            newItem->ItemUnion = new CEventTrace.Item.Union();
+            newItem->ItemType = (CEventTrace.ItemType) item.ItemType;
+            switch (item.ItemType)
+            {
+                case ItemType.Span:
+                    var spanItem = item.Span.GetValueOrDefault();
+                    newItem->ItemUnion.Span = new CEventTrace.Span();
+                    newItem->ItemUnion.Span.Id = ConvertSpanId(spanItem.Id);
+                    newItem->ItemUnion.Span.CauseCount = (uint) spanItem.Causes.Length;
+                    for (var i = 0; i < newItem->ItemUnion.Span.CauseCount; i++)
+                    {
+                        newItem->ItemUnion.Span.Causes[i] = ConvertSpanId(spanItem.Causes[i]);
+                        callback(newItem);
+                    }
+
+                    break;
+                case ItemType.Event:
+                    var eventItem = item.Event.GetValueOrDefault();
+                    ConvertEvent(eventItem, internalEvent =>
+                    {
+                        newItem->ItemUnion.Event = internalEvent;
+                        callback(newItem);
+                    });
+
+                    break;
+                default:
+                    throw new NotSupportedException("Invalid Item Type provided.");
+            }
+        }
+
+        public static unsafe void ConvertEventTracerParameters(EventTracerParameters[] parameters, EventTracerParametersCallback callback)
+        {
+            var handles = new List<WrappedGcHandle>(parameters.Length);
+            ConvertTracerParameters(handles, parameters, out var internalParameters);
+
+            fixed (CEventTrace.EventTracerParameters* parameterBuffer = internalParameters)
+            {
+                callback(parameterBuffer, handles);
+            }
+        }
+
+        private static unsafe WrappedGcHandle ConvertTracerParameter(EventTracerParameters parameter, ref CEventTrace.EventTracerParameters internalParameters)
+        {
+            WrappedGcHandle wrappedParameterObject = new WrappedGcHandle(parameter);
+
+            internalParameters.UserData = (void*) wrappedParameterObject.Get();
+            internalParameters.TraceCallback = parameter.TraceCallback == null
+                ? IntPtr.Zero
+                : Marshal.GetFunctionPointerForDelegate(CallbackThunkDelegates.TraceCallbackThunkDelegate);
+
+            return wrappedParameterObject;
+        }
+
+        public static void FreeTracerParameterHandles(IEnumerable<WrappedGcHandle> tracerParameterHandles)
```

Comment:
Is this worth a function?

Reply:
Probably not, I think I can move the logic to inside the Dispose method of the EventTracer instead